File: Editorial/2924_Find_Champion_II/CPP/1_Approach_In_degree_Count.cpp
class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        // Initialize the indegree array to track the number of incoming edges
        // for each team
        vector<int> indegree(n, 0);
        // Store the indegree of each team.
        for (auto edge : edges) {
            indegree[edge[1]]++;
        }

        int champ = -1, champCount = 0;
        for (int i = 0; i < n; i++) {
            // If the team can be champion, store the number and count of such
            // teams.
            if (indegree[i] == 0) {
                champCount++;
                champ = i;
            }
        }

        // If more than one team can be a champion, return -1, otherwise return
        // the champion team number
        return champCount > 1 ? -1 : champ;
    }
};

File: Editorial/1501-1600/1557_Minimum_Number_of_Vertices_to_Reach_All_Nodes/Python3/1_Approach_In_Degree_Count.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        in_degree = [False] * n
        for _, node2 in edges:
            in_degree[node2] = True

        return [node for node in range(n) if not in_degree[node]]
